rdmo_zenodo/exports/base.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+from django.conf import settings
+from django.shortcuts import reverse
+from django.utils.translation import gettext_lazy as _
+
+from rdmo.projects.exports import Export
+from rdmo.services.providers import OauthProviderMixin
+
+logger = logging.getLogger(__name__)
+
+
+class BaseZenodoExportProvider(OauthProviderMixin, Export):
+
+    @property
+    def client_id(self):
+        return settings.ZENODO_PROVIDER['client_id']
+
+    @property
+    def client_secret(self):
+        return settings.ZENODO_PROVIDER['client_secret']
+
+    @property
+    def zenodo_url(self):
+        return settings.ZENODO_PROVIDER.get('zenodo_url', 'https://sandbox.zenodo.org').strip('/')
+
+    @property
+    def authorize_url(self):
+        return f'{self.zenodo_url}/oauth/authorize'
+
+    @property
+    def token_url(self):
+        return f'{self.zenodo_url}/oauth/token'
+
+    @property
+    def deposit_url(self):
+        return f'{self.zenodo_url}/api/records'
+
+    @property
+    def redirect_path(self):
+        return reverse('oauth_callback', args=['zenodo'])
```

Comment:
one part of the error 500 bug is in here, for the new publish plugin it needs to pass `['zenodo-publish']` as `args`.
But I was still getting an error from zenodo in the response `errors?error=invalid_request&error_description=Mismatching+redirect+URI.&state=...`
maybe interesting for @cpfaff if you try to debug this

Reply:
after adding the Redirect URI (`http://localhost:8000/services/oauth/zenodo-publish/callback/`) to the [zenodo application settings ](https://sandbox.zenodo.org/account/settings/applications/) it works actually again.  I'll push the fix in this branch.